core/src/main/java/com/alibaba/nacos/core/cluster/ServerMemberManager.java

```diff
@@ -307,6 +307,14 @@ synchronized boolean memberChange(Collection<Member> members) {
             
             if (!serverList.containsKey(address)) {
                 hasChange = true;
+                // If the cluster information in cluster.conf or address-server has been changed,
+                // while the corresponding nacos-server has not been started yet, the member's state
+                // should be set to DOWN. If the corresponding nacos-server has been started, the
+                // member's state will be set to UP after detection in a few seconds.
+                member.setState(NodeState.DOWN);
+            } else {
+                //fix issue # 4925
+                member.setState(serverList.get(address).getState());
```

Comment:
I don't think it's a right fix.

this method is update members, so we should use input member state I think.

Reply:
I don't think we could use input member state in this method 
`synchronized boolean memberChange(Collection<Member> members) {}`

This method is only called in three places：
1.    
![image](https://user-images.githubusercontent.com/53824889/109130382-9db5db80-778c-11eb-9d1b-490b35699223.png)

In this case, because **the member state is set to UP by default**,  whether it's from cluster.conf  or address-server.
![image](https://user-images.githubusercontent.com/53824889/109131158-83303200-778d-11eb-8e4a-f759bccec187.png)

If a member has been judged as down,  it will be reset to UP after this method. **So this issue 4925 occured**.

2. 
![image](https://user-images.githubusercontent.com/53824889/109130411-aad2ca80-778c-11eb-8ff2-4c3c6ba41372.png)

In this case, a joined member is not in the serverList. We are **not sure** if the joined member's process has started, so it should be set to DOWN at first, and set to UP after MemberInfoReportTask completed.

3.
![image](https://user-images.githubusercontent.com/53824889/109130465-ba521380-778c-11eb-8bf6-d9e774f03405.png)

This has no effect on the method memberLeave()

Reply:
I'm not sure whether it will affect other report or update states workflow. I think we need do some discussion.

@chuntaojun @shiyiyue1102 How do you think about?

Reply:
Not only the state value of the member, I think all the properties of the member should be based on the value in the serverlist. 
Otherwise, the data returned by the interface /nacos/v1/ns/operator/servers is always changing (be reset)，for the same reason as issue 4925.

Reply:
But some of other Member attributes should modified by other members‘ report.
So I think the final value should be updated from the inputs.

Reply:
this is temporary plan, finally we should figure out the remote data update and local data update to solve this problem

Reply:
> But some of other Member attributes should modified by other members‘ report.
> So I think the final value should be updated from the inputs.

There is no problem that  final value should be updated from the inputs, because this method 
memberChange(Collection<Member> members)
is not called when nacos-server received other members' report.
It seems that this method is only called when the cluster information is updated，I think the attribute value of member should be based on the value in nacos-server memory.